File: dota2-senate/solution.py
class Solution:
    def predictPartyVictory(self, senate: str) -> str:
        to_ban_d=0
        to_ban_r=0
        cr=senate.count('R')
        cd=senate.count('D')
        banned=set()
        while True:
            for i in range(len(senate)):
                if i in banned:
                    continue
                if senate[i]=='D':
                    if to_ban_d>0:
                        to_ban_d-=1
                        banned.add(i)
                        cd-=1
                    else:
                        to_ban_r+=1
                if senate[i]=='R':
                    if to_ban_r>0:
                        to_ban_r-=1
                        banned.add(i)
                        cr-=1
                    else:
                        to_ban_d+=1
                if cr<=to_ban_r:
                    return 'Dire'
                if cd<=to_ban_d:
                    return 'Radiant'

File: dota2-senate/solution.cpp
class Solution {
public:
    string predictPartyVictory(string senate) {
        int br=0;
        int bd=0;
        int cr=0;
        int cd=0;
        int n=senate.size();
        for(int i=0;i<n;i++)
            if(senate[i]=='D')
                cd++;
            else
                cr++;
        while(1)
            for(int i=0;i<n;i++){
                if(senate[i]=='D'){
                    if(bd>0){
                        senate[i]='X';
                        bd--;
                        cd--;
                    }
                    else{
                        br++;
                    }
                }
                if(senate[i]=='R'){
                    if(br>0){
                        senate[i]='X';
                        br--;
                        cr--;
                    }
                    else{
                        bd++;
                    }
                }
                if(cr<br)
                    return "Dire";
                if(cd<bd)
                    return "Radiant";
            }
    
    }
};